rbus: prevent overflow when reading timeouts

Permanence-AI-Coder opened the pull request:
The changes in this PR were automatically generated by the Permanence AI Coder and reviewed by @jweese and @dnovick.

In each of the `rbusHandle_Fetch*Timeout` functions, there is a possibility that the integer read from the file is too large, causing an overflow and incorrect result for the timeout. This change introduces a helper function `rbusHandle_FetchTimeoutFromFile` and calls it from each of the four functions. As an enhancement, the helper function uses a float instead of the old implementation's integer to allow fractional timeouts. In the case that no data was read, that the float was too big (`ERANGE`), that the float was negative, or that multiplication would cause an overflow, the helper function returns a provided default value.

This PR complies with RDK LLM usage requirements.

on-behalf-of: @permanence-ai <github-ai@permanence.ai>

karuna2git commented on src/rbus/rbus_handle.c:
```diff
@@ -308,7 +308,11 @@ uint32_t rbusHandle_FetchSetMultiTimeout(rbusHandle_t handle)
     }
     if (timeout > 0)
+    {
+        if ((uint32_t)timeout > UINT32_MAX / 1000)
```
Change looks good; But,
1. Exact logic of retrieving timeout value from file is present above also. This check has to be added above as well.
2. There is bug (P2) already. If the timeout in the file is above `INT32_MAX` then the `atoi` will consider that as -ve and ignore the read timeout.
3. `Note: floating numbers case, it takes the whole number; but thats okey I guess`

jweese replied:
Thank you for the feedback. We could not find the P2 ticket on the RDK jira, so we decided to rerun with your floating-point feedback. Please re-review and advise if this PR should include all four change sites or if you would like a separate PR for each of the four.

karuna2git replied:
>So we decided to rerun with your floating-point feedback.
Thank you. Its developers (your) choice. The floating point is just one of the option that I shared. If there is another approach, we can consider that.

>PR should include all four change sites 
Please have all 4.

Thanks,

karuna2git commented on src/rbus/rbus_handle.c:
```diff
@@ -304,11 +305,19 @@ uint32_t rbusHandle_FetchSetMultiTimeout(rbusHandle_t handle)
+            if((errno == ERANGE) || (timeout > ((float)UINT32_MAX/1000.0f)))
+            {
+                fclose(fp);
+                return UINT32_MAX;
```
Its contradicting the approach.

When invalid input is given we ignore read timeout and use the default timeout `handle->timeoutValues.setMultiTimeout;` (Ref: line 321).

So, here at line 314, we must return default timeout is the appropriate approach. right?

jweese replied:
Thanks, understood. We will regenerate these changes with the following behavior: if any input is invalid (including the potential overflow), we will return the default as in L321. We will update this PR with changes to the four sites.

Permanence-AI-Coder committed:
rbus: prevent overflow when reading timeouts

The changes in this PR were automatically generated by the Permanence AI
Coder and reviewed by @jweese and @dnovick.

In each of the `rbusHandle_Fetch*Timeout` functions, there is a
possibility that the integer read from the file is too large, causing an
overflow and incorrect result for the timeout. This change introduces a
helper function `rbusHandle_FetchTimeoutFromFile` and calls it from each
of the four functions. As an enhancement, the helper function uses a
float instead of the old implementation's integer to allow fractional
timeouts. In the case that no data was read, that the float was too big
(`ERANGE`), that the float was negative, or that multiplication would
cause an overflow, the helper function returns a provided default value.

on-behalf-of: @permanence-ai <github-ai@permanence.ai>
